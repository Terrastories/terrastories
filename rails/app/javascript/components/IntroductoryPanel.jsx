import React from "react";
import PropTypes from 'prop-types';

const IntroductoryPanel = (props) => {
  const { numberOnePath, numberTwoPath } = props;

  return (
    <React.Fragment>
      <span className="card--nav-intro">{I18n.t("intro.question")}</span>
      <p>{I18n.t("intro.explanation")}</p>
      <span className="card--nav-intro">{I18n.t("usage.question")}</span>
      <div className="card--nav-usage">
        <div className="first-info">
          <div className="first-img">
            <img src={numberOnePath} alt="First" />
          </div>
          <p>{I18n.t("usage.category_explanation")}</p>
        </div>
        <div>
          <div className="second-img">
            <img src={numberTwoPath} alt="Second" />
          </div>  
          <p>{I18n.t("usage.option_explanation")}</p>
        </div>
      </div>
      <p>{I18n.t("usage.explore_explanation")}</p>
    </React.Fragment>
  )
}

IntroductoryPanel.propTypes = {
  numberOnePath: PropTypes.string,
  numberTwoPath: PropTypes.string
};

export default IntroductoryPanel;