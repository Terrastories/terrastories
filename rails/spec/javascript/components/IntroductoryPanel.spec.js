import React from 'react';
import { shallow } from 'enzyme';
import IntroductoryPanel from '../../../app/javascript/components/IntroductoryPanel';

describe('IntroductoryPanel Component', () => {
  let numberOnePath = 'first.svg';
  let numberTwoPath = 'second.svg';
  let wrapper;

  beforeEach(() => {
    global.I18n = require('./__mocks__/I18n.mock.js'),
    wrapper = shallow(
                <IntroductoryPanel 
                  numberOnePath={numberOnePath} 
                  numberTwoPath={numberTwoPath} 
                />);
  })

  it('Displays correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').find('.card--nav-intro')).toHaveLength(2);
    expect(wrapper.find('div').find('.card--nav-usage')).toHaveLength(1);
    expect(wrapper.find('div').find('.first-info')).toHaveLength(1);
    expect(wrapper.find('div').find('.first-img')).toHaveLength(1);
    expect(wrapper.find('div').find('.second-img')).toHaveLength(1);  
    expect(wrapper.find('img').first().prop('src')).toEqual('first.svg');
    expect(wrapper.find('img').last().prop('src')).toEqual('second.svg');
  })

  it('Renders the localized text', () => {
    expect(I18n.t).toHaveBeenCalledWith('intro.question');
    expect(I18n.t).toHaveBeenCalledWith('usage.question');
    expect(I18n.t).toHaveBeenCalledWith('usage.category_explanation');
    expect(I18n.t).toHaveBeenCalledWith('usage.option_explanation');
    expect(I18n.t).toHaveBeenCalledWith('usage.explore_explanation');
  })
})