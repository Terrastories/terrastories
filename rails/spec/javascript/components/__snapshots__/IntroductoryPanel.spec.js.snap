// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroductoryPanel Component Displays correctly 1`] = `
<Fragment>
  <span
    className="card--nav-intro"
  >
    intro.question in local translation
  </span>
  <p>
    intro.explanation in local translation
  </p>
  <span
    className="card--nav-intro"
  >
    usage.question in local translation
  </span>
  <div
    className="card--nav-usage"
  >
    <div
      className="first-info"
    >
      <div
        className="first-img"
      >
        <img
          alt="First"
          src="first.svg"
        />
      </div>
      <p>
        usage.category_explanation in local translation
      </p>
    </div>
    <div>
      <div
        className="second-img"
      >
        <img
          alt="Second"
          src="second.svg"
        />
      </div>
      <p>
        usage.option_explanation in local translation
      </p>
    </div>
  </div>
  <p>
    usage.explore_explanation in local translation
  </p>
</Fragment>
`;
